bin/internal/update_dart_sdk.sh

```diff
@@ -56,20 +56,29 @@ if [ ! -f "$ENGINE_STAMP" ] || [ "$ENGINE_VERSION" != `cat "$ENGINE_STAMP"` ]; t
     echo
     exit 1
   }
-  >&2 echo "Downloading Dart SDK from Flutter engine $ENGINE_VERSION..."
 
-  # On x64 stdout is "uname -m: x86_64"
-  # On arm64 stdout is "uname -m: aarch64, arm64_v8a"
-  case "$(uname -m)" in
-    x86_64)
+  OS="$(uname -s)"
+  # `uname -m` may be running in Rosetta mode, instead query sysctl
+  if [ "$OS" = 'Darwin' ]; then
+    ARCH="arm64"
+    # This will exit 1 if the CPU arch is not ARM64
```

Comment:
On x64 it will either exit with a failure code, or it will exit successfully and print `0` (I actually don't see that anymore, but I did see that working on https://github.com/flutter/flutter/pull/67970, though it may have been a beta version of macOS):
```
$ sysctl hw.optional.arm64
sysctl: unknown oid 'hw.optional.arm64'
$ echo $?
1
```
On arm64 it will print `1` and exit successfully.

Reply:
argh, we need to parse the output then?

Reply:
at least it looks like we can do `sysctl -n hw.optional.arm64` https://github.com/pytorch/pytorch/blob/e7ca62be08f96b07efc1271baf9b02d4e7be226a/cmake/Modules/FindARM.cmake#L44

It's annoying we have to ALSO check the exit value though

Reply:
If I recall the last time I looked at this on old versions of macOS it exited 1, on new versions of macOS on x64 it exited successfully and printed `0`, and on arm64 it printed `1`.  However on latest macOS on my x64 machine it's exiting failure with `sysctl: unknown oid 'hw.optional.arm64'`.  So maybe that's not the behavior anymore.

Reply:
Updated this to mirror the logic of https://github.com/flutter/flutter/pull/67970